torchtune/modules/transforms/tokenizers/_gpt2.py

```diff
@@ -0,0 +1,232 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+import json
+import re
+from functools import lru_cache
+from typing import List
+
+from torchtune.modules.transforms.tokenizers._utils import BaseTokenizer
+
+
+@lru_cache()
+def bytes_to_unicode() -> dict:
+    """
+    Returns list of utf-8 byte and a mapping to unicode strings. We specifically avoids mapping to whitespace/control
+    characters the bpe code barfs on.
+
+    Original paper: https://cdn.openai.com/better-language-models/language_models_are_unsupervised_multitask_learners.pdf
+
+    This is standard implementation based on:
+    https://github.com/huggingface/transformers/blob/6b550462139655d488d4c663086a63e98713c6b9/src/transformers/models/gpt2/tokenization_gpt2.py#L36
+    """
+    bs = (
+        list(range(ord("!"), ord("~") + 1))
+        + list(range(ord("¡"), ord("¬") + 1))
+        + list(range(ord("®"), ord("ÿ") + 1))
+    )
+    cs = bs[:]
+    n = 0
+    for b in range(2**8):
+        if b not in bs:
+            bs.append(b)
+            cs.append(2**8 + n)
+            n += 1
+    cs = [chr(n) for n in cs]
+    return dict(zip(bs, cs))
+
+
+def get_pairs(word: tuple) -> set:
+    """
+    Args:
+        word (tuple): Word is represented as tuple of symbols (symbols being variable-length strings).
+
+    Returns:
+        set of symbol pairs in a word.
+    """
+    pairs = set()
+    prev_char = word[0]
+    for char in word[1:]:
+        pairs.add((prev_char, char))
+        prev_char = char
+    return pairs
+
+
+class GPT2BaseTokenizer(BaseTokenizer):
+    """
+    A lightweight version of the basic GPT2Tokenizer.
+
+    Args:
+        vocab_path (str): Path to vocab.json file.
+        merges_path (str): Path to merges.txt file.
+        errors (str): Paradigm to follow when decoding.
```

Comment:
What are the acceptable values here? Looking at the docstring it's not clear to me how this is used